attest/verifier/src/quote.rs

```diff
@@ -0,0 +1,394 @@
+            sign_type == self.0
+        } else {
+            false
+        }
+    }
+}
+
+/// A [`Verify<Quote>`] implementation that will check if the XEID matches
+/// expectations.
+#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct XeidVerifier(u32);
+
+impl Verify<Quote> for XeidVerifier {
+    fn verify(&self, quote: &Quote) -> bool {
+        quote.xeid().map(|xeid| xeid == self.0).unwrap_or(false)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use mc_sgx_types::sgx_basename_t;
+    use mc_util_encodings::FromBase64;
+
+    const BASE64_QUOTE: &str = include_str!("../data/test/quote_ok.txt");
+    const BASE64_QUOTE2: &str = include_str!("../data/test/quote_configuration_needed.txt");
+
+    /// When the quote contains the basename we're expecting
+    #[test]
+    fn basename_success() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let verifier = Kind::from(quote.basename().expect("Could not read basename"));
+
+        assert!(verifier.verify(&quote));
+    }
+
+    /// When the quote does not contain the basename we're expecting
+    #[test]
+    fn basename_fail() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let basename = sgx_basename_t { name: [0u8; 32] };
+        let verifier = Kind::from(Basename::from(basename));
+
+        assert!(!verifier.verify(&quote));
+    }
+
+    /// When the quote matches what we're expecting
+    #[test]
+    fn quote_contents_success() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let verifier = Kind::from(quote.clone());
+
+        assert!(verifier.verify(&quote));
+    }
+
+    /// When the report does not contain the EPID group ID we're expecting
+    #[test]
+    fn quote_contents_fail() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let verifier = Kind::from(
+            Quote::from_base64(BASE64_QUOTE2)
+                .expect("Could not parse other quote from base64 file"),
+        );
+
+        assert!(!verifier.verify(&quote));
+    }
+
+    /// When the report contains the EPID group ID we're expecting
+    #[test]
+    fn epid_group_id_success() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let verifier = Kind::from(quote.epid_group_id().expect("Could not read EPID Group ID"));
+
+        assert!(verifier.verify(&quote));
+    }
+
+    /// When the report does not contain the EPID group ID we're expecting
+    #[test]
+    fn epid_group_id_fail() {
+        let quote =
+            Quote::from_base64(BASE64_QUOTE).expect("Could not parse quote from base64 file");
+        let epid_group_id = [0u8; 4];
```

Comment:
Can we turn these magic numbers into constants with descriptive names?

Reply:
These functions will be updated in part 3, where verifiers return `Option<Error>` instead of `bool`, I can take a look at that then, but I'd like to keep the amount of code changes in this PR to a minimum.